### PR TITLE
solution hw-spring-intro

### PR DESCRIPTION
**Pull request by RomanHreits**

### CODE REVIEW

**Review by aleksandr-hrankin — APPROVED** [review]

**Comment by aleksandr-hrankin** [comment]
Nice Job)

File: src/main/java/spring/intro/dao/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    public List<User> getUsers() {
+        try {
+            Session session = sessionFactory.openSession();
```
Comment (Sofasmile) [inline src/main/java/spring/intro/dao/UserDaoImpl.java]: where do we close the session?

File: src/main/java/spring/intro/dao/UserDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            return session.createQuery("from User", User.class).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get users from DB!", e);
```
Comment (Sofasmile) [inline src/main/java/spring/intro/dao/UserDaoImpl.java]: let's use DataProcessingException here too))

File: src/main/java/spring/intro/dao/UserDaoImpl.java (hunk 1 above)
Comment (RomanHreits) [reply src/main/java/spring/intro/dao/UserDaoImpl.java]: done

File: src/main/java/spring/intro/dao/UserDaoImpl.java (hunk 2 above)
Comment (RomanHreits) [reply src/main/java/spring/intro/dao/UserDaoImpl.java]: done))

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**